app/utils/client/lib/roomTypes.js

```diff
@@ -38,23 +38,30 @@ export const roomTypes = new class RocketChatRoomTypes extends RoomTypesCommon {
 			rid: roomId,
 		}).count() > 0;
 	}
-	readOnly(rid, user) {
+	readOnly(roomId, user) {
 		const fields = {
 			ro: 1,
+			t: 1,
 		};
 		if (user) {
 			fields.muted = 1;
 		}
 		const room = ChatRoom.findOne({
-			_id: rid,
+			_id: roomId,
 		}, {
 			fields,
 		});
+
+		const roomType = room.t;
+		if (room && roomType && this.roomTypes[roomType] && this.roomTypes[roomType].readOnly) {
```

Comment:
I dont think you need to check this `room` as you already made `const roomType = room.t;` or change the tests place